pkg/network/cluster_config.go

```diff
@@ -137,6 +137,11 @@ func StatusFromOperatorConfig(operConf *operv1.NetworkSpec, oldStatus *configv1.
 		status.NetworkType = string(operConf.DefaultNetwork.Type)
 	}
 
+	// Set networkType to empty string when preparing migration. So networkType value in spec can be consumed by MCO.
+	if migrationPrepare {
+		status.NetworkType = ""
```

Comment:
We can't do this. This is a public API; arbitrary components may be relying on the value of this field, for arbitrary purposes, and they might break if it is suddenly unset.

We may need to add a field to the config indicating that migration is in progress, so MCO can respond to that. (Maybe MCO could just read the annotation? That seems kind of bad though. If it's an actual API between CNO and MCO, then it should be actual API, not an annotation.)

Reply:
When the migration starts, the cluster will be in a maintenance state. The cluster shall be considered as out-of-service until the whole process finishes. We don't guarantee any of the cluster components can behave as normal during this period. So as long as the MCO can work as expected, I suppose it should be fine. WDYT?